nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -584,6 +589,51 @@ def _integrate_original_checkpoint_data(self, checkpoint: Dict[str, Any]) -> Dic
 
         return checkpoint
 
+    def _drop_optimizer_states(self, filepath: Union[str, Path], storage_options: Optional[Any] = None) -> None:
```

Comment:
most of the code below conceptually fits into ModelCheckpoint class (except for state dict construction, see [here](https://github.com/NVIDIA/NeMo/pull/10241/files#r1736080010))

I would suggest moving it all there except for this part which I guess can stay at `save_checkpoint` and we can just decide whether we want to include optimizer state in `storage_options`
 ```
            checkpoint['optimizer_states'] = [None]
            checkpoint['state_dict'] = OrderedDict([])
            checkpoint['sharded_state_dict'] = self.lightning_module.sharded_state_dict()

            # Save the checkpoint without optimizer states
            self.checkpoint_io.save_checkpoint(checkpoint, checkpoint_path, storage_options=storage_options)

            if HAVE_MODELOPT and hasattr(self.lightning_module, "get_model_module_list"):
                save_sharded_modelopt_state(
                    self.lightning_module.get_model_module_list(),
                    checkpoint_path,
                    self.unwrapped_checkpoint_io.save_sharded_strategy,
                    prefix="model.",
                )
```